conan/tools/system/package_manager.py

```diff
@@ -0,0 +1,212 @@
+        self._arch = self._conanfile.settings_build.get_safe('arch') \
+            if self._conanfile.context == CONTEXT_BUILD else self._conanfile.settings.get_safe('arch')
+        self._arch_names = {}
+        self._arch_separator = ""
+
+    def get_default_tool(self):
+        os_name = platform.system()
+        if os_name in ["Linux", "FreeBSD"]:
+            import distro
+            os_name = distro.id() or os_name
+        elif os_name == "Windows" and self._conanfile.conf["tools.microsoft.bash:subsystem"] == "msys2":
+            os_name = "msys2"
+        manager_mapping = {"apt-get": ["Linux", "ubuntu", "debian"],
+                           "yum": ["pidora", "scientific", "xenserver", "amazon", "oracle", "amzn",
+                                   "almalinux"],
+                           "dnf": ["fedora", "rhel", "centos", "mageia"],
+                           "brew": ["Darwin"],
+                           "pacman": ["arch", "manjaro", "msys2"],
+                           "choco": ["Windows"],
+                           "zypper": ["opensuse", "sles"],
+                           "pkg": ["freebsd"],
+                           "pkgutil": ["Solaris"]}
+        for tool, distros in manager_mapping.items():
+            if os_name in distros:
+                return tool
+
+    def get_package_name(self, package):
+        # TODO: should we only add the arch if cross-building?
+        if self._arch in self._arch_names:
+            return "{}{}{}".format(package, self._arch_separator,
+                                   self._arch_names.get(self._arch))
+        return package
+
+    @property
+    def sudo_str(self):
+        sudo = "sudo " if self._sudo else ""
+        askpass = "-A " if self._sudo and self._sudo_askpass else ""
+        return "{}{}".format(sudo, askpass)
+
+    def run(self, method, *args, **kwargs):
+        if self._active_tool == self.__class__.tool_name:
+            return method(*args, **kwargs)
+
+    def install(self, *args, **kwargs):
+        return self.run(self._install, *args, **kwargs)
+
+    def update(self, *args, **kwargs):
+        return self.run(self._update, *args, **kwargs)
+
+    def check(self, *args, **kwargs):
+        return self.run(self._check, *args, **kwargs)
+
+    def _install(self, packages, update=False, check=True, **kwargs):
+        if update:
+            self.update()
+
+        if check:
+            packages = self.check(packages)
+
+        if self._mode == self.mode_check and packages:
+            raise ConanException("System requirements: '{}' are missing but can't install "
+                                 "because tools.system.package_manager:mode is '{}'."
+                                 "Please update packages manually or set "
+                                 "'tools.system.package_manager:mode' "
+                                 "to '{}'".format(", ".join(packages),
```

Comment:
```suggestion
            raise ConanException("System requirements: '{}' are missing but can't install "
                                 "because tools.system.package_manager:mode is '{}'."
                                 "Please update system packages manually or set in your profile [conf] section"
                                 "'tools.system.package_manager:mode={}'".format(", ".join(packages),
```